### PR TITLE
GEODE-9037: Do not expose unsupported commands in Geode compatibility with Redis

### PR DESCRIPTION
**Pull request by jhutchison**
-update handling of unsupported commands

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Review by jdeppe-pivotal — CHANGES_REQUESTED** [review]

File: geode-redis/src/main/java/org/apache/geode/redis/internal/GeodeRedisServer.java [hunk 1]
```diff
@@ -46,34 +45,25 @@
...
 public class GeodeRedisServer {
+
   /**
    * The default Redis port as specified by their protocol, {@code DEFAULT_REDIS_SERVER_PORT}
    */
   public static final int DEFAULT_REDIS_SERVER_PORT = 6379;
-
-  public static final String ENABLE_REDIS_UNSUPPORTED_COMMANDS_PARAM =
-      "enable-redis-unsupported-commands";
+  public static final String ENABLE_UNSUPPORTED_COMMANDS_SYSTEM_PROPERTY =
```
[the remark this answers is not in the document]
Comment (jhutchison) [reply geode-redis/src/main/java/org/apache/geode/redis/internal/GeodeRedisServer.java:54]: ok-  will change back

File: geode-redis/src/commonTest/java/org/apache/geode/redis/GeodeRedisServerRule.java [hunk 2]
```diff
@@ -40,15 +41,21 @@ public GeodeRedisServerRule() {
     cacheFactory.set(LOCATORS, "");
   }
 
+  public GeodeRedisServerRule(boolean enableUnsupportedCommands) {
+    cacheFactory = new CacheFactory();
+    cacheFactory.set(LOG_LEVEL, "warn");
+    cacheFactory.set(MCAST_PORT, "0");
+    cacheFactory.set(LOCATORS, "");
```
[the remark this answers is not in the document]
Comment (jhutchison) [reply geode-redis/src/commonTest/java/org/apache/geode/redis/GeodeRedisServerRule.java:48]: thx.  will change

File: geode-redis/src/main/java/org/apache/geode/redis/internal/GeodeRedisServer.java [hunk 3]
```diff
@@ -142,16 +124,14 @@ protected Long getDataStoreBytesInUseForDataRegion() {
...
+  @VisibleForTesting
+  public void setAllowUnsupportedCommandsSystemProperty(boolean allowUnsupportedCommands) {
+    System.setProperty(ENABLE_UNSUPPORTED_COMMANDS_SYSTEM_PROPERTY,
+        String.valueOf(allowUnsupportedCommands));
   }
 
-  private void logUnsupportedCommandWarning() {
-    logger.warn("Unsupported commands enabled. Unsupported commands have not been fully tested.");
+  public boolean allowUnsupportedCommands() {
+    return Boolean.valueOf(System.getProperty("enable-unsupported-commands", "false"));
```
[the remark this answers is not in the document]
Comment (jhutchison) [reply geode-redis/src/main/java/org/apache/geode/redis/internal/GeodeRedisServer.java:134]: thanks.  changed

File: geode-redis/src/main/java/org/apache/geode/redis/internal/GeodeRedisServer.java [hunk 4]
```diff
@@ -142,16 +124,14 @@ protected Long getDataStoreBytesInUseForDataRegion() {
...
   }
 
-  /**
-   * Precedence of the internal property overrides the global system property.
-   */
-  public boolean allowUnsupportedCommands() {
-    return (boolean) regionProvider.getConfigRegion()
-        .getOrDefault(ENABLE_REDIS_UNSUPPORTED_COMMANDS_PARAM, ENABLE_REDIS_UNSUPPORTED_COMMANDS);
+  @VisibleForTesting
+  public void setAllowUnsupportedCommandsSystemProperty(boolean allowUnsupportedCommands) {
```
[the remark this answers is not in the document]
Comment (jhutchison) [reply geode-redis/src/main/java/org/apache/geode/redis/internal/GeodeRedisServer.java:128]: yeah, I think it's an expression of my uncertainty about how we had it implemented.  it's just setting the field now (which I feel better about).

**Review by jdeppe-pivotal — APPROVED** [review]

**Review by rhoughton-pivot — APPROVED** [review]

**Review by dickcav — APPROVED** [review]

**Review by nonbinaryprogrammer — APPROVED** [review]

**Pull request merged**